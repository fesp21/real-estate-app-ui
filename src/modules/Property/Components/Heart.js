import React, {Component, PropTypes} from "react";
import {View, StyleSheet, StatusBar, Text,TouchableHighlight} from "react-native";
import Icon from "react-native-vector-icons/FontAwesome";

export default class Heart extends Component {

  static propTypes = {
    // loadFilterScene:PropTypes.func.isRequired
    handleFavoritePress:PropTypes.func.isRequired
  };

  render() {
    const {handleFavoritePress,isFavorited} = this.props;
    return (
      <View style={styles.container}>
        <TouchableHighlight onPress={()=>{handleFavoritePress()}} underlayColor="transparent">
          <Icon name={isFavorited ? 'heart': 'heart-o'} size={25} style={[styles.icon]} />
        </TouchableHighlight>
      </View>
    );
  }
}

const styles =  StyleSheet.create({
  container : {
    justifyContent:'flex-end',
    alignItems:'center'
  },
  title:{
    color:'white',
    fontSize:15,
    paddingLeft:5
  },
  icon: {
    width:25,
    height:25,
    color:'red',
  }
});