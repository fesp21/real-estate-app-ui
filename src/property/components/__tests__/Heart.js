import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import Heart from "./../Heart";

test('renders heart', () => {
  const tree = renderer.create(
    <Heart
      isFavorited={true}
      handleFavoritePress={()=>{}}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
