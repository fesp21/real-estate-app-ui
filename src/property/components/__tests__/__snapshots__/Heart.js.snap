exports[`test renders heart 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "flex-end",
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 25,
          },
          Array [
            Object {
              "color": "red",
              "height": 25,
              "width": 25,
            },
          ],
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      ï€„
    </Text>
  </View>
</View>
`;
